### ISSUE TITLE
Build failure aganist Hyprland f590505daf90dfb4059289a906863b59663e70a6 on NixOS

### ISSUE THREAD

**Issue by nonetrix** [Duckonaut/split-monitor-workspaces#53]
## Output
```
error: builder for '/nix/store/75idw6vr1klhqz67ki3zs6pzrg0vnbs8-split-monitor-workspaces-0.1.drv' failed with exit code 1;
       last 10 log lines:
       > Running phase: buildPhase
       > build flags: -j16
       > [1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
       > FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o
       > g++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/nix/store/ivn3k9533as5kalgkwgnpj44c3p4w232-hyprland-0.36.0+date=2024-02-29_f590505-dev/include -I/nix/store/ivn3k9533as5kalgkwgnpj44c3p4w232-hyprland-0.36.0+date=2024-02-29_f590505-dev/include/hyprland/protocols -I/nix/store/ivn3k9533as5kalgkwgnpj44c3p4w232-hyprland-0.36.0+date=2024-02-29_f590505-dev/include/hyprland/wlroots -I/nix/store/mk3hpjywv0hkkn2bwplcb9lwvzsv2j0i-pixman-0.43.2/include/pixman-1 -I/nix/store/sg7sijswn5a2mb9ycjbir22mrbgib70n-libdrm-2.4.120-dev/include -I/nix/store/sg7sijswn5a2mb9ycjbir22mrbgib70n-libdrm-2.4.120-dev/include/libdrm -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++23 -DWLR_USE_UNSTABLE -fPIC -MD -MQ libsplit-monitor-workspaces.so.p/src_main.cpp.o -MF libsplit-monitor-workspaces.so.p/src_main.cpp.o.d -o libsplit-monitor-workspaces.so.p/src_main.cpp.o -c ../src/main.cpp
       > ../src/main.cpp:6:10: fatal error: hyprland/src/helpers/Workspace.hpp: No such file or directory
       >     6 | #include <hyprland/src/helpers/Workspace.hpp>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/75idw6vr1klhqz67ki3zs6pzrg0vnbs8-split-monitor-workspaces-0.1.drv'.
error: 1 dependencies of derivation '/nix/store/k3r89axw1v071jz16sg9kcdbm229q2nf-hm_hyprhyprland.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/js74508dddahrz90gyfyc7471x8rvaaz-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/jf62s6agh4iwwmyy507b4ma1d5rx3kvg-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/49iwsbk7jjyy8shw1l7j8bvnha01hrnh-unit-home-manager-noah.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5dsvq3j1w9zr6pd5in2fr04wvagr2gd9-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/69910rvpsmpsp5v9i5jg1y0fb9k21afv-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/klz4ykv0r7fncbbmph1r2ga0lx4fwpyg-nixos-system-nixos-24.05.20240228.9099616.drv' failed to build
```
## Full log
```
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/75idw6vr1klhqz67ki3zs6pzrg0vnbs8-split-monitor-workspaces-0.1.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/bfvsaw766asr1zn15hbz65bgf7drpy13-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
mesonConfigurePhase flags: --prefix=/nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1 --libdir=/nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces>
The Meson build system
Version: 1.3.2
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: g++ (gcc 13.2.0 "g++ (GCC) 13.2.0")
C++ linker for the host machine: g++ ld.bfd 2.40
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: YES (/nix/store/492n7ylp2f9zn0k3vghv1snypgkzznc5-pkg-config-wrapper-0.29.2/bin/pkg-config) 0.29.2
Run-time dependency hyprland found: YES 0.36.0
Run-time dependency pixman-1 found: YES 0.43.2
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

split-monitor-workspaces 0.1

  User defined options
    auto_features: enabled
    bindir       : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/bin
    buildtype    : plain
    includedir   : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/include
    infodir      : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/share/info
    libdir       : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/lib
    libexecdir   : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/libexec
    localedir    : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/share/locale
    mandir       : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/share/man
    prefix       : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1
    sbindir      : /nix/store/kbym3pimvyn615czb5qp9cbj60qkl9f1-split-monitor-workspaces-0.1/sbin
    wrap_mode    : nodownload

Found ninja-1.11.1 at /nix/store/3dj1s1bhxanq3f0nffwf786l71g1lcn6-ninja-1.11.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: -j16
[1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o
g++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/nix/store/ivn3k9533as5kalgkwgnpj44c3p4w232-hyprland-0.36.0+date=2024-02-29_f590505-dev/include -I/nix/store/ivn3k9533as5ka>
../src/main.cpp:6:10: fatal error: hyprland/src/helpers/Workspace.hpp: No such file or directory
    6 | #include <hyprland/src/helpers/Workspace.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```
## Flake.lock
```
{
  "nodes": {
    "home-manager": {
      "inputs": {
        "nixpkgs": [
          "nixpkgs"
        ]
      },
      "locked": {
        "lastModified": 1709204054,
        "narHash": "sha256-U1idK0JHs1XOfSI1APYuXi4AEADf+B+ZU4Wifc0pBHk=",
        "owner": "nix-community",
        "repo": "home-manager",
        "rev": "2f3367769a93b226c467551315e9e270c3f78b15",
        "type": "github"
      },
      "original": {
        "owner": "nix-community",
        "repo": "home-manager",
        "type": "github"
      }
    },
    "hyprland": {
      "inputs": {
        "hyprland-protocols": "hyprland-protocols",
        "hyprlang": "hyprlang",
        "nixpkgs": "nixpkgs",
        "systems": "systems",
        "wlroots": "wlroots",
        "xdph": "xdph"
      },
      "locked": {
        "lastModified": 1709243510,
        "narHash": "sha256-e47KWaUGQc+JLUABYdG0eCeqY1fFf1xGvakpKcdOP9c=",
        "owner": "hyprwm",
        "repo": "Hyprland",
        "rev": "f590505daf90dfb4059289a906863b59663e70a6",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "Hyprland",
        "type": "github"
      }
    },
    "hyprland-protocols": {
      "inputs": {
        "nixpkgs": [
          "hyprland",
          "nixpkgs"
        ],
        "systems": [
          "hyprland",
          "systems"
        ]
      },
      "locked": {
        "lastModified": 1691753796,
        "narHash": "sha256-zOEwiWoXk3j3+EoF3ySUJmberFewWlagvewDRuWYAso=",
        "owner": "hyprwm",
        "repo": "hyprland-protocols",
        "rev": "0c2ce70625cb30aef199cb388f99e19a61a6ce03",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "hyprland-protocols",
        "type": "github"
      }
    },
    "hyprlang": {
      "inputs": {
        "nixpkgs": [
          "hyprland",
          "nixpkgs"
        ],
        "systems": [
          "hyprland",
          "systems"
        ]
      },
      "locked": {
        "lastModified": 1708787654,
        "narHash": "sha256-7ACgM3ZuAhPqurXHUvR2nWMRcnmzGGPjLK6q4DSTelI=",
        "owner": "hyprwm",
        "repo": "hyprlang",
        "rev": "0fce791ba2334aca183f2ed42399518947550d0d",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "hyprlang",
        "type": "github"
      }
    },
    "nix-filter": {
      "locked": {
        "lastModified": 1693833173,
        "narHash": "sha256-hlMABKrGbEiJD5dwUSfnw1CQ3bG7KKwDV+Nx3bEZd7U=",
        "owner": "numtide",
        "repo": "nix-filter",
        "rev": "ac030bd9ba98e318e1f4c4328d60766ade8ebe8b",
        "type": "github"
      },
      "original": {
        "owner": "numtide",
        "repo": "nix-filter",
        "type": "github"
      }
    },
    "nixpkgs": {
      "locked": {
        "lastModified": 1708807242,
        "narHash": "sha256-sRTRkhMD4delO/hPxxi+XwLqPn8BuUq6nnj4JqLwOu0=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "73de017ef2d18a04ac4bfd0c02650007ccb31c2a",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "nixpkgs_2": {
      "locked": {
        "lastModified": 1709150264,
        "narHash": "sha256-HofykKuisObPUfj0E9CJVfaMhawXkYx3G8UIFR/XQ38=",
        "owner": "NixOS",
        "repo": "nixpkgs",
        "rev": "9099616b93301d5cf84274b184a3a5ec69e94e08",
        "type": "github"
      },
      "original": {
        "owner": "NixOS",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
    "nur": {
      "locked": {
        "lastModified": 1709248623,
        "narHash": "sha256-Zmo187XIeCGlGHRcczf5tboirJghYesHmX2Eug8CQRY=",
        "owner": "nix-community",
        "repo": "NUR",
        "rev": "14de3286e2081dbffaa7ebb595363227b6af2a74",
        "type": "github"
      },
      "original": {
        "owner": "nix-community",
        "repo": "NUR",
        "type": "github"
      }
    },
    "root": {
      "inputs": {
        "home-manager": "home-manager",
        "hyprland": "hyprland",
        "nixpkgs": "nixpkgs_2",
        "nur": "nur",
        "split-monitor-workspaces": "split-monitor-workspaces"
      }
    },
    "split-monitor-workspaces": {
      "inputs": {
        "hyprland": [
          "hyprland"
        ],
        "nix-filter": "nix-filter"
      },
      "locked": {
        "lastModified": 1709119202,
        "narHash": "sha256-3qcLLe+LqC1YAdqpKtU1COlxzb5uqpTMYgwmleS2oRg=",
        "owner": "Duckonaut",
        "repo": "split-monitor-workspaces",
        "rev": "d0012b8b0f764e32dd7b82f7a94b8c30197d7dc8",
        "type": "github"
      },
      "original": {
        "owner": "Duckonaut",
        "repo": "split-monitor-workspaces",
        "type": "github"
      }
    },
    "systems": {
      "locked": {
        "lastModified": 1689347949,
        "narHash": "sha256-12tWmuL2zgBgZkdoB6qXZsgJEH9LR3oUgpaQq2RbI80=",
        "owner": "nix-systems",
        "repo": "default-linux",
        "rev": "31732fcf5e8fea42e59c2488ad31a0e651500f68",
        "type": "github"
      },
      "original": {
        "owner": "nix-systems",
        "repo": "default-linux",
        "type": "github"
      }
    },
    "wlroots": {
      "flake": false,
      "locked": {
        "host": "gitlab.freedesktop.org",
        "lastModified": 1708558866,
        "narHash": "sha256-Mz6hCtommq7RQfcPnxLINigO4RYSNt23HeJHC6mVmWI=",
        "owner": "wlroots",
        "repo": "wlroots",
        "rev": "0cb091f1a2d345f37d2ee445f4ffd04f7f4ec9e5",
        "type": "gitlab"
      },
      "original": {
        "host": "gitlab.freedesktop.org",
        "owner": "wlroots",
        "repo": "wlroots",
        "rev": "0cb091f1a2d345f37d2ee445f4ffd04f7f4ec9e5",
        "type": "gitlab"
      }
    },
    "xdph": {
      "inputs": {
        "hyprland-protocols": [
          "hyprland",
          "hyprland-protocols"
        ],
        "hyprlang": [
          "hyprland",
          "hyprlang"
        ],
        "nixpkgs": [
          "hyprland",
          "nixpkgs"
        ],
        "systems": [
          "hyprland",
          "systems"
        ]
      },
      "locked": {
        "lastModified": 1708696469,
        "narHash": "sha256-shh5wmpeYy3MmsBfkm4f76yPsBDGk6OLYRVG+ARy2F0=",
        "owner": "hyprwm",
        "repo": "xdg-desktop-portal-hyprland",
        "rev": "1b713911c2f12b96c2574474686e4027ac4bf826",
        "type": "github"
      },
      "original": {
        "owner": "hyprwm",
        "repo": "xdg-desktop-portal-hyprland",
        "type": "github"
      }
    }
  },
  "root": "root",
  "version": 7
}
```

**Comment by John-Paul-R** [comment]
I believe this is a result of this Hyprland change: https://github.com/hyprwm/Hyprland/commit/b39dcfa497f84486569bf862092dfcfadbfd8747

**Comment by fecet** [comment]
It broken again with `hyprlang>4.2`

**Comment by jules-sommer** [comment]
Heyyy, just wanted to contribute some logs, obviously the issue is simply that the specific Workspace.hpp header file required during the build isn't being found for some reason ( I don't really have time to investigate and debug rn as to why that'd be, sorryyy :disappointed: but hope the logs are useful at the very least ). It does seem that the compiler is seeing all the build inputs specified in the Nix derivation but is just missing the one header file.

Ninja build logs:
```bash
.config/nixos on  fix-main [$✘!+?] took 26s 
❯ : nix log /nix/store/8izbsrawgv4dbh8fyfz8z4401hi8z403-split-monitor-workspaces-0.1.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/8izbsrawgv4dbh8fyfz8z4401hi8z403-split-monitor-workspaces-0.1.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/bfvsaw766asr1zn15hbz65bgf7drpy13-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
mesonConfigurePhase flags: --prefix=/nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspace>
The Meson build system
Version: 1.3.2
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: g++ (gcc 13.2.0 "g++ (GCC) 13.2.0")
C++ linker for the host machine: g++ ld.bfd 2.40
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES 
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of usi>
Found pkg-config: YES (/nix/store/492n7ylp2f9zn0k3vghv1snypgkzznc5-pkg-config-wrapper-0.29.2/bin/pkg-co>
Run-time dependency hyprland found: YES 0.36.0
Run-time dependency pixman-1 found: YES 0.43.2
Run-time dependency libdrm found: YES 2.4.120
Build dir: /build/source/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: g++ (gcc 13.2.0 "g++ (GCC) 13.2.0")
C++ linker for the host machine: g++ ld.bfd 2.40
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES 
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of usi>
Found pkg-config: YES (/nix/store/492n7ylp2f9zn0k3vghv1snypgkzznc5-pkg-config-wrapper-0.29.2/bin/pkg-co>
Run-time dependency hyprland found: YES 0.36.0
Run-time dependency pixman-1 found: YES 0.43.2
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

split-monitor-workspaces 0.1

  User defined options
    auto_features: enabled
    bindir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/bin
    buildtype    : plain
    includedir   : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/include
    infodir      : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/info
    libdir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/lib
    libexecdir   : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/libexec
    localedir    : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/locale
    mandir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/man
    prefix       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1
    sbindir      : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/sbin
    wrap_mode    : nodownload
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/bfvsaw766asr1zn15hbz65bgf7drpy13-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
mesonConfigurePhase flags: --prefix=/nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspace>
The Meson build system
Version: 1.3.2
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: g++ (gcc 13.2.0 "g++ (GCC) 13.2.0")
C++ linker for the host machine: g++ ld.bfd 2.40
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES 
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of usi>
Found pkg-config: YES (/nix/store/492n7ylp2f9zn0k3vghv1snypgkzznc5-pkg-config-wrapper-0.29.2/bin/pkg-co>
Run-time dependency hyprland found: YES 0.36.0
Run-time dependency pixman-1 found: YES 0.43.2
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

split-monitor-workspaces 0.1

  User defined options
    auto_features: enabled
    bindir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/bin
    buildtype    : plain
    includedir   : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/include
    infodir      : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/info
    libdir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/lib
    libexecdir   : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/libexec
    localedir    : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/locale
    mandir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/man
Build dir: /build/source/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: g++ (gcc 13.2.0 "g++ (GCC) 13.2.0")
C++ linker for the host machine: g++ ld.bfd 2.40
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES 
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of usi>
Found pkg-config: YES (/nix/store/492n7ylp2f9zn0k3vghv1snypgkzznc5-pkg-config-wrapper-0.29.2/bin/pkg-co>
Run-time dependency hyprland found: YES 0.36.0
Run-time dependency pixman-1 found: YES 0.43.2
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

split-monitor-workspaces 0.1

  User defined options
    auto_features: enabled
    bindir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/bin
    buildtype    : plain
    includedir   : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/include
    infodir      : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/info
    libdir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/lib
    libexecdir   : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/libexec
    localedir    : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/locale
    mandir       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/share/man
    prefix       : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1
    sbindir      : /nix/store/55gx7hppc0ybgn2d37pji4y2mknqpjf5-split-monitor-workspaces-0.1/sbin
    wrap_mode    : nodownload

Found ninja-1.11.1 at /nix/store/3dj1s1bhxanq3f0nffwf786l71g1lcn6-ninja-1.11.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
@nix { "action": "setPhase", "phase": "buildPhase" }
Running phase: buildPhase
build flags: -j12
[1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o 
g++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/nix/store/l7ck9rqw64bj6wl2mq5k0m2lg98m6>
../src/main.cpp:6:10: fatal error: hyprland/src/helpers/Workspace.hpp: No such file or directory
    6 | #include <hyprland/src/helpers/Workspace.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
``` 

Nix flake rebuild:
```bash
.config/nixos on  fix-main [$✘!+?] took 1m30s 
❯ : flake switch .#ishottt --show-trace
info  Switching system configuration to .#ishottt
warning: $HOME ('/home/jules') is not owned by you, falling back to the one defined in the 'passwd' file ('/root')
warning: input 'flake-utils' has an override for a non-existent input 'nixpkgs'
building the system configuration...
warning: $HOME ('/home/jules') is not owned by you, falling back to the one defined in the 'passwd' file ('/root')
warning: input 'flake-utils' has an override for a non-existent input 'nixpkgs'
trace: warning: getExe: Package "bun-1.0.29" does not have the meta.mainProgram attribute. We'll assume that the main program has the same name for now, but this behavior is deprecated, because it leads to surprising errors when the assumption does not hold. If the package has a main program, please set `meta.mainProgram` in its definition to make this warning go away. Otherwise, if the package does not have a main program, or if you don't control its definition, use getExe' to specify the name to the program, such as lib.getExe' foo "bar".
[1/0/10 built] building etc-nix-registry.jsonwarning: $HOME ('/home/jules') is not owned by you, falling back to the one defined in the 'passwd' file ('/root')
error: builder for '/nix/store/8izbsrawgv4dbh8fyfz8z4401hi8z403-split-monitor-workspaces-0.1.drv' failed with exit code 1;
       last 10 log lines:
       > Running phase: buildPhase
       > build flags: -j12
       > [1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
       > FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o
       > g++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/nix/store/l7ck9rqw64bj6wl2mq5k0m2lg98m634r-hyprland-0.36.0+date=2024-03-08_e52d3fa-dev/include -I/nix/store/l7ck9rqw64bj6wl2mq5k0m2lg98m634r-hyprland-0.36.0+date=2024-03-08_e52d3fa-dev/include/hyprland/protocols -I/nix/store/l7ck9rqw64bj6wl2mq5k0m2lg98m634r-hyprland-0.36.0+date=2024-03-08_e52d3fa-dev/include/hyprland/wlroots -I/nix/store/mk3hpjywv0hkkn2bwplcb9lwvzsv2j0i-pixman-0.43.2/include/pixman-1 -I/nix/store/sg7sijswn5a2mb9ycjbir22mrbgib70n-libdrm-2.4.120-dev/include -I/nix/store/sg7sijswn5a2mb9ycjbir22mrbgib70n-libdrm-2.4.120-dev/include/libdrm -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++23 -DWLR_USE_UNSTABLE -fPIC -MD -MQ libsplit-monitor-workspaces.so.p/src_main.cpp.o -MF libsplit-monitor-workspaces.so.p/src_main.cpp.o.d -o libsplit-monitor-workspaces.so.p/src_main.cpp.o -c ../src/main.cpp
       > ../src/main.cpp:6:10: fatal error: hyprland/src/helpers/Workspace.hpp: No such file or directory
       >     6 | #include <hyprland/src/helpers/Workspace.hpp>
       >       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > compilation terminated.
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/8izbsrawgv4dbh8fyfz8z4401hi8z403-split-monitor-workspaces-0.1.drv'.
error: 1 dependencies of derivation '/nix/store/jw3kvpiwmwvk7x149lckw522plzy9py0-hm_hyprhyprland.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cdr4na7l4y4qk4zmbfyw28g71v10n0zf-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3hwbrx2w1hwk15y8iqr154rwzgsq9xl0-home-manager-generation.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2kfzy71n154whn9cblhwqqn77nny4b3q-unit-home-manager-jules.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1ii19j69qxhqraygkp4j07i7dcg5zzqn-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/y0j28zkh68n4jw9a38hb6sfwhs6f9cl2-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bvdm7y8s4z0b31gy1kfghjd5s7vd0wg0-nixos-system-ishottt-24.05.20240306.9df3e30.drv' failed to build
```

And just to give specific information on the versions of all relevant packages I am on:

```nix
[Resolved URL:]  git+file:///home/jules/_dev/.config?dir=nixos
[Locked URL:]    git+file:///home/jules/_dev/.config?dir=nixos
[Description:]   JulesOS NixOS configuration flake
[Path:]          /nix/store/71w20kvqsfmymhiwgfii0i4v12jl2zvz-source
[Revision:]      d47c4d9040b94de802dd5595003ac0cbd5d97fa0-dirty
[Last modified:] 2024-03-05 12:41:10
[Inputs:]
├───[home-manager]: github:nix-community/home-manager/417015af0dc2525557ab36528643c2d519ca4334 (2024-03-08 12:58:55)
│   └───[nixpkgs] follows input 'nixpkgs'
├───[hyprland]: github:hyprwm/Hyprland/e52d3fa8522ff83813a9006d4bf46b3057f4a4cc (2024-03-08 02:24:44)
│   ├───[hyprland-protocols]: github:hyprwm/hyprland-protocols/0c2ce70625cb30aef199cb388f99e19a61a6ce03 (2023-08-11 11:36:36)
│   │   ├───[nixpkgs] follows input 'hyprland/nixpkgs'
│   │   └───[systems] follows input 'hyprland/systems'
│   ├───[hyprlang]: github:hyprwm/hyprlang/f1db1a7e1faee2a5c67d03b6bd283da82eed3730 (2024-03-07 01:41:15)
│   │   ├───[nixpkgs] follows input 'hyprland/nixpkgs'
│   │   └───[systems] follows input 'hyprland/systems'
│   ├───[nixpkgs]: github:NixOS/nixpkgs/9df3e30ce24fd28c7b3e2de0d986769db5d6225d (2024-03-06 05:30:39)
│   ├───[systems]: github:nix-systems/default-linux/31732fcf5e8fea42e59c2488ad31a0e651500f68 (2023-07-14 15:19:09)
│   ├───[wlroots]: gitlab:wlroots/wlroots/0cb091f1a2d345f37d2ee445f4ffd04f7f4ec9e5 (2024-02-21 23:41:06)
│   └───[xdph]: github:hyprwm/xdg-desktop-portal-hyprland/2d2fb547178ec025da643db57d40a971507b82fe (2024-03-01 13:27:19)
│       ├───[hyprland-protocols] follows input 'hyprland/hyprland-protocols'
│       ├───[hyprlang] follows input 'hyprland/hyprlang'
│       ├───[nixpkgs] follows input 'hyprland/nixpkgs'
│       └───[systems] follows input 'hyprland/systems'
├───[hyprland-plugins]: github:hyprwm/hyprland-plugins/8d3a0701a26d51c9931b3583f93e7fb3fbcd5110 (2024-03-04 00:35:28)
│   └───[hyprland] follows input 'hyprland'
├───[nixpkgs]: github:nixos/nixpkgs/9df3e30ce24fd28c7b3e2de0d986769db5d6225d (2024-03-06 05:30:39)
├───[nixpkgs-unfree]: github:numtide/nixpkgs-unfree/1cfef288e8ddfb79fba9e11cc869ccf1bcfc01e6 (2024-02-27 17:29:31)
│   └───[nixpkgs] follows input 'nixpkgs'
├───[split-monitor-workspaces]: github:Duckonaut/split-monitor-workspaces/d0012b8b0f764e32dd7b82f7a94b8c30197d7dc8 (2024-02-28 11:20:02)
│   ├───[hyprland] follows input 'hyprland'
│   └───[nix-filter]: github:numtide/nix-filter/ac030bd9ba98e318e1f4c4328d60766ade8ebe8b (2023-09-04 13:12:53)
└───[stable]: github:nixos/nixpkgs/880992dcc006a5e00dd0591446fdf723e6a51a64 (2024-03-05 22:18:01)
```

**Comment by Duckonaut** [comment]
Fixed in #52
